tests/acceptance/stepDefinitions/webdavContext.js

```diff
@@ -18,8 +27,57 @@ Then('as {string} file/folder {string} should exist', function (userId, element)
         throw Error('file/folder should exist, but does not')
       }
     })
+}
+
+Then('as {string} file/folder {string} should exist', function (userId, element) {
+  return fileExists(userId, element)
 })
 
+Then('as {string} the last uploaded folder should exist', function (userId) {
+  return fileExists(userId, client.sessionId)
+})
+
+Then(
+  'as {string} the last uploaded folder should contain the following files inside the sub-folders:',
+  async function (user, files) {
+    files = files.raw().map(item => item[0])
+
+    const sessionId = client.sessionId
+    const response = await webdavHelper.propfind(
+      `/files/${user}/${sessionId}`,
+      user,
+      [],
+      2
+    )
+    const result = xml2js(response, { compact: true })
+
+    const elements = _.get(result, 'd:multistatus.d:response')
+    if (elements === undefined) {
+      throw new Error('Received unexpected response:\n' + result)
+    }
+
+    const uploadedFilesHref = elements
+      .map(elem => _.get(elem, 'd:href._text'))
+      .filter(item => item)
+
+    const regexToExtractBaseName = RegExp(
+      `/${sessionId}/upload[0-9]+file/(.*)`
+    )
```

Comment:
Cannot use named-group regex here. We seem to be running an old node version. I have opened an issue in #2345.